tests/console/yast2_lan.pm

```diff
@@ -64,26 +80,119 @@ sub run {
         }
     }
 
-    my $hostname = get_var('HOSTNAME', 'susetest');
-    my $domain   = "zq1.de";
-
     send_key "alt-s";    # open hostname tab
     assert_screen [qw(yast2_lan-hostname-tab dhcp-popup)];
     handle_dhcp_popup;
     send_key "tab";
     for (1 .. 15) { send_key "backspace" }
     type_string $hostname;
-    send_key "tab";
-    for (1 .. 15) { send_key "backspace" }
-    type_string $domain;
+    # on SLE15-SP1+ and Leap 15.1+ the domain field is dropped
+    if (is_sle('<=15') || is_leap('<=15.0')) {
+        send_key "tab";
+        for (1 .. 15) { send_key "backspace" }
+        type_string $domain;
+    }
     assert_screen 'test-yast2_lan-1';
+    close_yast2_lan;
 
-    send_key "alt-o";    # OK=>Save&Exit
-    wait_serial("yast2-lan-status-0", 180) || die "'yast2 lan' didn't finish";
+    # verify that hostname has been changed
+    assert_script_run "hostname | grep $hostname";
 
+    open_yast2_lan;
+    assert_screen [qw(yast2_lan dhcp-popup)], 90;
```

Comment:
I would say that verification of YaST lan opened should be inside function which responsible for opening it

Reply:
True, that makes sense from the semantic point of view. I opted for this solution because upon first opening there are more checks for pop ups etc., but I'll rework that.